### PR TITLE
Add resource for habitat build and publish

### PR DESCRIPTION
**Pull request by elliott-davis**

### CODE REVIEW

File: libraries/helpers.rb [hunk 1]
```diff
@@ -97,3 +97,26 @@ def project_secrets_exist?(secret_keys = [])
...
+def build_plan(origin, plan_dir, cwd)
+  execute 'build-plan' do
+    command "sudo -E #{hab_binary} studio" \
+            " -r #{hab_studio_path}" \
+            " build #{plan_dir}"
+    environment({'TERM' => 'vt100', 'HAB_ORIGIN' => origin})
+    cwd cwd
+    live_stream true
+  end
+end
```
Comment (jtimberman) [inline libraries/helpers.rb]: IMO we should just put this in the resource. See below.

File: resources/habitat.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+property :origin, String, required: true
+property :plan_dir, String, required: true
+property :cwd, String, required: true
+property :url, String
+property :artifact, String
+property :home_dir, String
+property :auth_token, String
+
+action :build do
+  build_plan(origin, plan_dir, cwd)
```
Comment (jtimberman) [inline resources/habitat.rb]: Move this from the library helper.

File: resources/habitat.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+end
+
+action :build_and_publish do
```
Comment (jtimberman) [inline resources/habitat.rb]: This should be `:publish`

File: resources/habitat.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  build_plan(origin, plan_dir, cwd)
```
Comment (jtimberman) [inline resources/habitat.rb]: Remove this.

File: recipes/publish.rb [hunk 5]
```diff
@@ -29,44 +29,19 @@
...
-      live_stream true
-      sensitive true
+    habitat project_name do
+      origin origin
+      plan_dir habitat_plan_dir
+      cwd node['delivery']['workspace']['repo']
+      home_dir delivery_workspace
+      auth_token project_secrets['habitat']['depot_token']
+      url node['habitat-build']['depot-url']
+      action :build_and_publish
```
Comment (jtimberman) [inline recipes/publish.rb]: With the other changes I'm suggesting, your action would look like this:

```ruby
action [:build, :publish]
```

This is idiomatic and expected behavior for Chef users. For example, commonly a `service` resource will be written like:

```ruby
service "food" do
  action [:enable, :start]
end
```

File: resources/habitat.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
+require_relative '../libraries/helpers'
+
+actions :build, :build_and_publish
```
Comment (schisamo) [inline resources/habitat.rb]: I don't think you need to enumerate all the actions, if so be sure to change `:build_and_publish` to `:publish`.

**Comment by elliott-davis** [comment]
comments addressed.

**Comment by schisamo** [comment]
Closing this now as we are going to enable Automate's GitHub v2 integration for this pipeline. We'll recreate the PR using `delivery review` shortly.

**Pull request closed without merging**